src/build/content/server.ts

```diff
@@ -116,36 +116,53 @@ export const copyNextServerCode = async (ctx: PluginContext): Promise<void> => {
       },
     )
 
-    await Promise.all(
-      paths.map(async (path: string) => {
-        const srcPath = join(srcDir, path)
-        const destPath = join(destDir, path)
-
-        // If this is the middleware manifest file, replace it with an empty
-        // manifest to avoid running middleware again in the server handler.
-        if (path === 'server/middleware-manifest.json') {
-          try {
-            await replaceMiddlewareManifest(srcPath, destPath)
-          } catch (error) {
-            throw new Error('Could not patch middleware manifest file', { cause: error })
-          }
+    const promises = paths.map(async (path: string) => {
+      const srcPath = join(srcDir, path)
+      const destPath = join(destDir, path)
 
-          return
+      // If this is the middleware manifest file, replace it with an empty
+      // manifest to avoid running middleware again in the server handler.
+      if (path === 'server/middleware-manifest.json') {
+        try {
+          await replaceMiddlewareManifest(srcPath, destPath)
+        } catch (error) {
+          throw new Error('Could not patch middleware manifest file', { cause: error })
         }
 
-        if (path === 'server/functions-config-manifest.json') {
-          try {
-            await replaceFunctionsConfigManifest(srcPath, destPath)
-          } catch (error) {
-            throw new Error('Could not patch functions config manifest file', { cause: error })
-          }
+        return
+      }
 
-          return
+      if (path === 'server/functions-config-manifest.json') {
+        try {
+          await replaceFunctionsConfigManifest(srcPath, destPath)
+        } catch (error) {
+          throw new Error('Could not patch functions config manifest file', { cause: error })
         }
 
-        await cp(srcPath, destPath, { recursive: true, force: true })
-      }),
-    )
+        return
+      }
+
+      await cp(srcPath, destPath, { recursive: true, force: true })
+    })
+
+    // this is different node_modules than ones handled by `copyNextDependencies`
+    // this is under the standalone/.next folder (not standalone/node_modules or standalone/<some-workspace/node_modules)
+    // and started to be created by Next.js in some cases in next@16.1.0-canary.3
+    // this node_modules is artificially created and doesn't have equivalent in the repo
+    // so we only copy it, without additional symlinks handling
+    if (existsSync(join(srcDir, 'node_modules'))) {
+      const filter = ctx.constants.IS_LOCAL ? undefined : nodeModulesFilter
+      const src = join(srcDir, 'node_modules')
+      const dest = join(destDir, 'node_modules')
+      await cp(src, dest, {
+        recursive: true,
+        verbatimSymlinks: true,
+        force: true,
+        filter,
+      })
+    }
```

Comment:
This part is fix for next@canary change to ensure the "hashed" artificial `node_modules` are included in NF (see first part of description for more details including first screenshot)